src/test/java/org/sagebionetworks/bridge/models/CriteriaUtilsTest.java

```diff
@@ -266,74 +268,71 @@ public void validateDataGroupSetsCannotBeNull() {
                 .addAll(minAppVersions.keySet()).addAll(maxAppVersions.keySet()).build(); }
         };
         
-        Errors errors = Validate.getErrorsFor(criteria);
+        Errors errors = mock(Errors.class);
         CriteriaUtils.validate(criteria, EMPTY_SET, EMPTY_SET, errors);
-        assertEquals(errors.getFieldErrors("allOfGroups").get(0).getCode(), "cannot be null");
-        assertEquals(errors.getFieldErrors("noneOfGroups").get(0).getCode(), "cannot be null");
-        assertEquals(errors.getFieldErrors("allOfStudyIds").get(0).getCode(), "cannot be null");
-        assertEquals(errors.getFieldErrors("noneOfStudyIds").get(0).getCode(), "cannot be null");
+        verify(errors).rejectValue("allOfGroups", "cannot be null");
+        verify(errors).rejectValue("noneOfGroups", "cannot be null");
+        verify(errors).rejectValue("allOfStudyIds", "cannot be null");
+        verify(errors).rejectValue("noneOfStudyIds", "cannot be null");
     }
     
     @Test
     public void validateDataGroupCannotBeWrong() {
         Criteria criteria = getCriteria().allOfGroups(ImmutableSet.of("group1"))
                 .noneOfGroups(ImmutableSet.of("group2")).build();
         
-        Errors errors = Validate.getErrorsFor(criteria);
+        Errors errors = mock(Errors.class);
         CriteriaUtils.validate(criteria, ImmutableSet.of("group3"), EMPTY_SET, errors);
-        assertEquals(errors.getFieldErrors("allOfGroups").get(0).getCode(), "'group1' is not in enumeration: group3");
-        assertEquals(errors.getFieldErrors("noneOfGroups").get(0).getCode(), "'group2' is not in enumeration: group3");
+        verify(errors).rejectValue("allOfGroups", "'group1' is not in enumeration: group3");
+        verify(errors).rejectValue("noneOfGroups", "'group2' is not in enumeration: group3");
     }
     
     @Test
     public void validateDataGroupNotBothRequiredAndProhibited() {
         Criteria criteria = getCriteria().allOfGroups(ImmutableSet.of("group1", "group2", "group3"))
                 .noneOfGroups(ImmutableSet.of("group2", "group3")).build();
         
-        Errors errors = Validate.getErrorsFor(criteria);
+        Errors errors = mock(Errors.class);
         CriteriaUtils.validate(criteria, ImmutableSet.of("group1","group2","group3","group4"), EMPTY_SET, errors);
         // It's a set so validate without describing the order of the groups in the error message
-        assertTrue(errors.getFieldErrors("allOfGroups").get(0).getCode().contains("includes these excluded data groups: "));
-        assertTrue(errors.getFieldErrors("allOfGroups").get(0).getCode().contains("group2"));
-        assertTrue(errors.getFieldErrors("allOfGroups").get(0).getCode().contains("group3"));
+        verify(errors).rejectValue("allOfGroups", "includes these excluded data groups: group2, group3");
```

Comment:
ImmutableSet (from Google's Guava library) is unusual for a set because the implementation guarantees that elements will be returned in the same order they are specified in the factory method.